common/dpcpp/slab_hash.hpp

```diff
@@ -88,16 +158,42 @@ template <typename K, typename T, typename Hash> class SlabHashTable {
 
       sycl::group_barrier(_gr);
     }
-
     return _ans;
   }
 
 private:
+  void alloc_node(sycl::device_ptr<SlabNode<std::pair<K, T>>> &src) {
```

Comment:
Function signature is a bit confusing, could you just return a pointer to a newly allocated node? Why does it have a parameter at all?

Reply:
in the case of allocation to the tail of the list, you need the `src` to be `prev - >next`, and if this is the first list for the node, you need the `src` to be `(lists + hash) - >root`. in order not to write two almost identical functions, I wanted to generalize them to both cases in this way, but maybe there is a more intelligent option..
and why return a pointer? if for `_iter`, then it is really possible to do this, but for `src` it seems not, because it needs to be in the lock zone